components/frontend/src/metric/MetricDetails.js

```diff
@@ -171,32 +171,24 @@ export function MetricDetails({
             { iconName: "linegraph" },
         ),
     )
-    if (measurements.length > 0) {
-        lastMeasurement.sources.forEach((source) => {
-            const reportSource = metric.sources[source.source_uuid]
-            if (!reportSource) {
-                return
-            } // source was deleted, continue
-            const nrEntities = source.entities?.length ?? 0
-            if (nrEntities === 0) {
-                return
-            } // no entities to show, continue
-            const sourceName = getSourceName(reportSource, dataModel)
-            panes.push(
-                tabPane(
-                    sourceName,
-                    <SourceEntities
-                        report={report}
-                        metric={metric}
-                        metric_uuid={metric_uuid}
-                        source={source}
-                        reload={measurementsReload}
-                    />,
-                    { image: <Logo logo={reportSource.type} alt={sourceName} /> },
-                ),
-            )
-        })
-    }
+    Object.entries(metric.sources).forEach(([source_uuid, source]) => {
```

Comment:
Interesting simplification, does that come from an assumption we can make regarding the input data or was it just not so optimal before?

Reply:
Or was this logic moved to SourceEntities.js ?

Reply:
It's simpler here because MetricDetails can just unconditionally render the SourceEntities tabs and the logic for dealing with missing measurement and/or entities is moved to the  SourceEntities component. Basically in the form of guard clauses that check whether the measurements are still being loaded, loading failed, or whether there are no entities at all, and if so, render an error or warning message.